backend/src/schema/resolvers/reports.spec.js

```diff
@@ -139,55 +198,120 @@ describe('file a report on a resource', () => {
-                },
-              },
-              errors: undefined,
+              })
+
+              const reportsCypherQueryResponse = await instance.cypher(reportsCypherQuery, {
+                resourceId: 'abusive-user-id',
+                currentUserId: authenticatedUser.id,
+              })
+              expect(reportsCypherQueryResponse.records).toHaveLength(1)
+              const [reportProperties] = reportsCypherQueryResponse.records.map(
+                record => record.get('report').properties,
+              )
+              expect(reportProperties).toMatchObject({ rule: 'latestReviewUpdatedAtRules' })
+            })
+
+            describe('with overtaken disabled from resource in disable property', () => {
+              it('disable is false', async () => {
+                await mutate({
+                  mutation: fileReportMutation,
+                  variables: { ...variables, resourceId: 'abusive-user-id' },
+                })
+
+                const reportsCypherQueryResponse = await instance.cypher(reportsCypherQuery, {
+                  resourceId: 'abusive-user-id',
+                  currentUserId: authenticatedUser.id,
+                })
+                expect(reportsCypherQueryResponse.records).toHaveLength(1)
+                const [reportProperties] = reportsCypherQueryResponse.records.map(
+                  record => record.get('report').properties,
+                )
+                expect(reportProperties).toMatchObject({ disable: false })
+              })
+
+              it('disable is true', async () => {
+                // first time filling a report to enable a moderator the disable the resource
+                await mutate({
+                  mutation: fileReportMutation,
+                  variables: { ...variables, resourceId: 'abusive-user-id' },
+                })
+                authenticatedUser = await moderator.toJson()
+                await mutate({
+                  mutation: reviewMutation,
+                  variables: {
+                    resourceId: 'abusive-user-id',
+                    disable: true,
+                    closed: true,
+                  },
+                })
+                authenticatedUser = await currentUser.toJson()
+                // second time filling a report to see if the "disable is true" of the resource is overtaken
+                await mutate({
+                  mutation: fileReportMutation,
+                  variables: { ...variables, resourceId: 'abusive-user-id' },
+                })
+
+                const reportsCypherQueryResponse = await instance.cypher(reportsCypherQuery, {
+                  resourceId: 'abusive-user-id',
+                  currentUserId: authenticatedUser.id,
+                })
+                expect(reportsCypherQueryResponse.records).toHaveLength(1)
+                const [reportProperties] = reportsCypherQueryResponse.records.map(
+                  record => record.get('report').properties,
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2020-02-21T11:39:35Z" title="Friday, February 21st 2020, 12:39:35 pm +01:00">Feb 21, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -139,55 +198,120 @@ describe('file a report on a resource', () => {
           it('which belongs to resource', async () => {
             await expect(
               mutate({
-                mutation: reportMutation,
+                mutation: fileReportMutation,
                 variables: { ...variables, resourceId: 'abusive-user-id' },
               }),
             ).resolves.toMatchObject({
               data: {
                 fileReport: {
-                  id: expect.any(String),
+                  reportId: expect.any(String),
+                  resource: {
+                    name: 'abusive-user',
+                  },
                 },
               },
               errors: undefined,
             })
           })
 
-          it('creates only one report for multiple reports on the same resource', async () => {
+          it('only one report for multiple reports on the same resource', async () => {
             const firstReport = await mutate({
-              mutation: reportMutation,
+              mutation: fileReportMutation,
               variables: { ...variables, resourceId: 'abusive-user-id' },
             })
             authenticatedUser = await otherReportingUser.toJson()
             const secondReport = await mutate({
-              mutation: reportMutation,
+              mutation: fileReportMutation,
               variables: { ...variables, resourceId: 'abusive-user-id' },
             })
-            expect(firstReport.data.fileReport.id).toEqual(secondReport.data.fileReport.id)
+
+            expect(firstReport.data.fileReport.reportId).toEqual(
+              secondReport.data.fileReport.reportId,
+            )
           })
 
-          it('returns the rule for how the report was decided', async () => {
-            await expect(
-              mutate({
-                mutation: reportMutation,
+          describe('report properties are set correctly', () => {
+            const reportsCypherQuery =
+              'MATCH (resource:User {id: $resourceId})<-[:BELONGS_TO]-(report:Report {closed: false})<-[filed:FILED]-(user:User {id: $currentUserId}) RETURN report'
+
+            it('with the rule for how the report will be decided', async () => {
+              await mutate({
+                mutation: fileReportMutation,
                 variables: { ...variables, resourceId: 'abusive-user-id' },
-              }),
-            ).resolves.toMatchObject({
-              data: {
-                fileReport: {
-                  rule: 'latestReviewUpdatedAtRules',
-                },
-              },
-              errors: undefined,
+              })
+
+              const reportsCypherQueryResponse = await instance.cypher(reportsCypherQuery, {
+                resourceId: 'abusive-user-id',
+                currentUserId: authenticatedUser.id,
+              })
+              expect(reportsCypherQueryResponse.records).toHaveLength(1)
+              const [reportProperties] = reportsCypherQueryResponse.records.map(
+                record => record.get('report').properties,
+              )
+              expect(reportProperties).toMatchObject({ rule: 'latestReviewUpdatedAtRules' })
+            })
+
+            describe('with overtaken disabled from resource in disable property', () => {
+              it('disable is false', async () => {
+                await mutate({
+                  mutation: fileReportMutation,
+                  variables: { ...variables, resourceId: 'abusive-user-id' },
+                })
+
+                const reportsCypherQueryResponse = await instance.cypher(reportsCypherQuery, {
+                  resourceId: 'abusive-user-id',
+                  currentUserId: authenticatedUser.id,
+                })
+                expect(reportsCypherQueryResponse.records).toHaveLength(1)
+                const [reportProperties] = reportsCypherQueryResponse.records.map(
+                  record => record.get('report').properties,
+                )
+                expect(reportProperties).toMatchObject({ disable: false })
+              })
+
+              it('disable is true', async () => {
+                // first time filling a report to enable a moderator the disable the resource
+                await mutate({
+                  mutation: fileReportMutation,
+                  variables: { ...variables, resourceId: 'abusive-user-id' },
+                })
+                authenticatedUser = await moderator.toJson()
+                await mutate({
+                  mutation: reviewMutation,
+                  variables: {
+                    resourceId: 'abusive-user-id',
+                    disable: true,
+                    closed: true,
+                  },
+                })
+                authenticatedUser = await currentUser.toJson()
+                // second time filling a report to see if the "disable is true" of the resource is overtaken
+                await mutate({
+                  mutation: fileReportMutation,
+                  variables: { ...variables, resourceId: 'abusive-user-id' },
+                })
+
+                const reportsCypherQueryResponse = await instance.cypher(reportsCypherQuery, {
+                  resourceId: 'abusive-user-id',
+                  currentUserId: authenticatedUser.id,
+                })
+                expect(reportsCypherQueryResponse.records).toHaveLength(1)
+                const [reportProperties] = reportsCypherQueryResponse.records.map(
+                  record => record.get('report').properties,
```


This `.properties` can be removed if you use `{.*}` in the cypher query.